src/example-types/results.js

```diff
@@ -24,43 +31,72 @@ module.exports = {
+    if (context.include) searchObj._source.includes = context.include
+    if (context.exclude) searchObj._source.excludes = context.exclude
 
-    if (context.include || context.exclude) result._source = {}
-    if (context.include) result._source.includes = context.include
-    if (context.exclude) result._source.excludes = context.exclude
+    // Global schema highlight configuration
+    let schemaHighlight =
+      _.getOr(true, 'highlight', context) &&
+      _.cloneDeep(schema.elasticsearch.highlight)
+    // Specific search highlight override
+    let searchHighlight = _.isPlainObject(context.highlight)
+      ? context.highlight
+      : {}
+    let resultColumns = context.include
 
-    let highlight =
-      _.getOr(true, 'highlight', context) && schema.elasticsearch.highlight
-    let inlineAliases = _.getOr({}, 'inlineAliases', highlight)
-    let showOtherMatches = _.getOr(false, 'showOtherMatches', context)
+    // Highlighting starts with defaults in the schema first
+    if (schemaHighlight) {
+      let showOtherMatches = _.getOr(false, 'showOtherMatches', context)
+      let schemaInline = _.getOr([], 'inline', schemaHighlight)
+      let schemaInlineAliases = _.getOr({}, 'inlineAliases', schemaHighlight)
 
-    if (highlight) {
+      // Concat the search specific override fields with the schema `inline` so we have them as targets for highlight replacement
+      F.setOn(
+        'inline',
+        _.concat(schemaInline, _.keys(searchHighlight.fields)),
+        schemaHighlight
+      )
       // Convert the highlight fields from array to an object map
       let fields = _.flow(
         _.pick(['inline', 'additionalFields']), // Get the highlight fields we will be working with
         _.values,
         _.flatten,
-        _.concat(_.values(inlineAliases)), // Include the provided field aliases if any
-        _.uniq(),
+        _.concat(_.values(schemaInlineAliases)), // Include the provided field aliases if any
+        _.uniq,
         arrayToHighlightsFieldMap, // Convert the array to object map so we can simply _.pick again
         filtered =>
           showOtherMatches
             ? filtered // Highlight on all fields specified in the initial _.pick above
             : _.pick(context.include, filtered) // Only highlight on the fields listed in the context include section
-      )(highlight)
+      )(schemaHighlight)
+
+      // Setup the DEFAULT highlight config object with the calculated fields above
+      // and merge with the search specific config
+      F.extendOn(
+        searchObj,
+        _.merge(
+          {
+            highlight: { ...defaultHighlightingConfig, fields },
+          },
+          {
+            highlight: searchHighlight,
+          }
+        )
+      )
+
+      // Make sure the search specific overrides are part of the context include.
+      // This way they do not have to be added manually. All that is needed is the highlight config
+      resultColumns = _.flow(
```

Comment:
This is a nice improvement - let's add a test case for it (in lieu of real documentation 😅)